### PR TITLE
P value groups

### PR DESCRIPTION
**Pull request by FelixMay**
Hi @dmcglinn and @rueuntal , here is my next version, now with the p-values at the group-level again.
I hope this version can now be used for the analysis that Jon et al. would like to do for the concept paper and for the ESA workshop?
The bootstrap CIs are not yet in, but  plan to include them as option only. But due to their problems (implicit rarefaction) this is of minor importance I think?

### CODE REVIEW

File: R/mobr_boxplots.R [hunk 1]
```diff
@@ -125,24 +124,105 @@ get_pval <- function(rand, obs, n_samples)
...
+      out$S  = rowSums(abund_mat > 0)  
+   }  
+   
+   # Rarefied richness ---------------------------------------------------------
+   if ("S_rare" %in% index){  
+      
+      out$S_rare <- list()
+      
+      n_rare <- floor(n_rare)
+      if (is.na(n_rare)){
```
Comment (rueuntal) [inline R/mobr_boxplots.R]: I'm guessing that `n_rare` is a vector. When `is.na()` is applied to a vector, it returns a binary vector of TRUE/FALSE, and the if statement may not work as intended.

File: R/mobr_boxplots.R [hunk 2]
```diff
@@ -231,59 +351,38 @@ get_mob_stats = function(mob_in,
       out$samples$S_rare <- list()
       
       n_rare_samples <- floor(n_rare_samples)
-      if (is.na(n_rare_samples)){   
+      if (!is.numeric(n_rare_samples) | is.na(n_rare_samples)){
```
Comment (rueuntal) [inline R/mobr_boxplots.R]: The same comment of applying `is.na()` to vector also applies here. Or maybe I'm misunderstanding what you are trying to do?

**Comment by rueuntal** [comment]
Hi @FelixMay - The implementation looks great! I have a question about applying `is.na()` to vectors (which could just be my misunderstanding). Once that's resolved I think this can be merged.

**Comment by FelixMay** [comment]
Hi @rueuntal , the idea is to set n_rare to the minimum sample size when there is no value provided. I now used `is.null` for that purpose plus `n_rare_samples = NULL` and `n_rare_groups = NULL` as default values. Maybe this is more explicit? When you have other suggestions how to code this mixture between usign defaults and error checking let me know.

**Comment by rueuntal** [comment]
Ah, I see. `n_rare_groups` and `n_rare_samples` can either be undefined (`NA` or `NULL`), or defined by user. If it's undefined, then the function assigns value to it. Right?

I don't think there's anything wrong with your previous implementation, it does what it supposed to. However if these values are input as vectors then there's a warning saying that `is.na` is being applied to more than one value, which doesn't look to be best practice. One way to remove that warning is to use `any(is.na())` instead (or `any(is.null())` in current implementation). But that's probably just an aesthetic issue.

Feel free to change it or not, and once you are happy feel free to merge!

**Pull request merged**